plugins/modules/jenkins_credential.py

```diff
@@ -0,0 +1,831 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) Ansible Project
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = r"""
+---
+module: jenkins_credential
+short_description: Manage Jenkins credentials and domains via API
+version_added: 1.0.0
+description:
+  - This module allows managing Jenkins credentials and domain scopes via Jenkins HTTP API.
+  - You can create, update, and delete different credential types such as username/password, secret text, SSH key, certificates, GitHub App, and scoped domains.
+  - For scoped credentials, it supports hostname, hostname:port, path, and scheme-based restrictions.
```

Comment:
```suggestion
  - For scoped credentials, it supports V(hostname), V(hostname:port), V(path), and scheme-based restrictions.
```
See https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_documenting.html#semantic-markup-within-module-documentation for more info.